src/index.js

```diff
@@ -20,18 +18,16 @@ export default class I18n {
 
   setLocale (locale: string): void {
     this.locale = locale
-
-    if (numeralLanguages[locale]) {
-      numeral.locales[locale] = numeralLanguages[locale]
-      numeral.locale(locale)
-    }
   }
 
   /**
-   * Leverages `numeral` for number formatting
+   * Leverages Intl.NumberFormat for currency formatting
    */
-  formatNumber (format: string, number: number): string {
-    return numeral(number).format(format)
+  formatNumber (number: number, style?: string, currency?: string): string {
```

Comment:
```javascript
type BaseFormat = {
  number: number
}
type FormatCurrency = BaseFormat & {
  number: number
  style: 'currency',
  currency: 'EUR' | 'DOLLAR'
}
type OtherStyles = BaseFormat & { 
  style: 'decimal' | 'percent' | 'unit' 
}
type FormatOptions = BaseFormat | OtherStyles | CurrencyStyle
formatNumber (number: number, { style, currency }) {
}
```

[Repel example](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoALgTwA4FMwBicATgLYCG6AwgK7HG4B2AxpmALxgDeqYYAzlhi4AXGADkzOgxaZxAGl5gp9JqzHiAogFUASuLAAfCQBEA8gBkLAQX2oAvhhz4z6ABa5iAZSG5+HbjAlQUxhDQATXGYASwoYA2NxPGJmJnQEiRpGaPSgxyw8QhIKdAA5GlIAI08zbHRouEZ-TiIySlpVWSMwVw9vX35UKCzmesawKGLKcqrPMAAKRgrq4jEl2eJ5MDg6hqaAfjFWkpmV2rGmgEpuJWZGwQmp9HO95u3d+6NjLkc+RlwkABzeCVcgwAB0AElGOgIadPMdKPNxH4FO8LvxLuDJm10Itlp5Lg5UEMnvDiPMAMyUrZcAS+CJRWJg8T2Ik4k4EinU2n00KiCTJVIw1nsslcqk0wIhMKZbLpNmk3HkyW8mUCyTSNRyLYqGTqCQ6fRgNlAA)

Reply:
@andresgutgon you are making a lot of assumptions here. Typescript itself marks `style` and `currency` as `string` for instance:

https://github.com/Microsoft/TypeScript/blob/master/src/lib/es5.d.ts#L4216

Reply:
I wouldn't complicate it for now

Reply:
Ok but we're not going to support all currencies in the world. Only a subset. Also this project use Flow isn't it?

With that typing we diferienciate between 3 ways of using this method.
1. No options
2. Format a 'decimal' | 'percent' | 'unit' 
3. Format a currency (the once we support)

But as I said totally fine to ignore this if you think doesn't make sense

Reply:
LOOOOL
![image](https://user-images.githubusercontent.com/49499/68576360-39bfb480-046e-11ea-8e6e-96d7db7dc4dc.png)


Eso no vale. Switch to TS in the same PR 😂